src/shared/storage/index.js

```diff
@@ -148,7 +170,27 @@ class Account {
      * @param {string} name
      */
     static delete(name) {
-        realm.write(() => realm.delete(Account.getObjectForId(name)));
+        realm.write(() => {
+            const accountsBeforeDeletion = Account.getDataAsArray();
+            const accountForDeletion = find(accountsBeforeDeletion, { name });
```

Comment:
Would it be easier to simply do this?

```js
const accountForDeletion = Account.getObjectForId(name);
```

Reply:
[getDataAsArray](https://github.com/iotaledger/trinity-wallet/blob/7da5732886cea2d3640ded8d81bf50449b027188/src/shared/storage/index.js#L73) already gets all the accounts in realm db. I think it's better to just find the account (to delete) rather than making another query to realm db.

Reply:
Ah, I see